### PR TITLE
Add documentation on remote recovery

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #35975. It adds documentation on the remote recovery
process. Additionally, it adds documentation about the various settings
that can impact the process.

### CODE REVIEW

**Review by ywelsch — COMMENTED** [review]
Thanks @tbrooks8. I've left mainly minor comments, this is looking very good already.

**Review by ywelsch — APPROVED** [review]

**Comment by lcawl** [comment]
@tbrooks8 Were you intending this new page to be added to the existing CCR documentation in the Stack Overview?  If so, I added a commit that puts it here:
image
If that's not what you intended, let me know.

File: docs/reference/ccr/getting-started.asciidoc [hunk 1]
```diff
@@ -253,6 +253,11 @@ PUT /server-metrics-copy/_ccr/follow?wait_for_active_shards=1
 
 //////////////////////////
 
+The follower index is bootstrapped using the <<remote-recovery, remote recovery>>
+process. The remote recovery process transfers the existing Lucene segment files
+from the leader to the follower. When the remote recovery process is complete,
+the index following will be initiated.
```
Comment (lcawl) [inline docs/reference/ccr/getting-started.asciidoc]:
```suggestion
the index following begins.
```
Or "is initiated".

File: docs/reference/ccr/getting-started.asciidoc [hunk 2]
```diff
@@ -253,6 +253,11 @@ PUT /server-metrics-copy/_ccr/follow?wait_for_active_shards=1
 
 //////////////////////////
 
+The follower index is bootstrapped using the <<remote-recovery, remote recovery>>
```
Comment (lcawl) [inline docs/reference/ccr/getting-started.asciidoc]: I find "bootstrapped" unclear. Could we another verb like "initialized", "prepared", "loaded", ...?

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
+[role="xpack"]
+[testenv="platinum"]
+[[remote-recovery]]
+== Remote recovery
+
+Remote recovery is the process used to build a new copy of a shard on a follower
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
 The _remote recovery_ process builds a new copy of a shard on a follower
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
+[role="xpack"]
+[testenv="platinum"]
+[[remote-recovery]]
+== Remote recovery
+
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]: I think you need a sentence or two of context at the beginning of this page. For example:

```suggestion
When you create a follower index, you cannot use it until it is fully initialized. 
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Remote recovery is the process used to build a new copy of a shard on a follower
+node by copying data from the primary shard in the leader cluster. {es} uses this
+remote recovery process to bootstrap a follower index using the data from the
+leader index. This allows the follower to receive a copy of the current state of
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
leader index. This process provides the follower with a copy of the current state of
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+the leader index, even if a complete history of changes is not available on the
+leader due to Lucene segment merging.
+
+Remote recovery is a network intensive process that transfers all of the Lucene
+segment files from the leader cluster to the follower cluster. The follower
+requests that a recovery session be initiated on the primary shard in the leader
+cluster. The follower then requests file chunks concurrently from the leader. By
+default, the the process concurrently requests `5` large `1mb` file chunks as remote
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
default, the process concurrently requests `5` large `1mb` file chunks as remote
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+recovery is designed to support leader and follower clusters with high network
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
behavior is designed to support leader and follower clusters with high network
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+latency between them.
+
+Information about an in-progress remote recovery can be obtained using the
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
You can obtain information about an in-progress remote recovery by using the
```

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+{ref}/cat-recovery.html[recovery API] on the follower cluster. Remote recoveries are implemented
+using the {ref}/modules-snapshots.html[snapshot and restore] infrastructure. This means that
+on-going remote recoveries will be labelled as type `snapshot` in the recovery api.
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]:
```suggestion
on-going remote recoveries are labelled as type `snapshot` in the recovery API.
```

Comment (lcawl) [reply docs/reference/ccr/remote-recovery.asciidoc]: Is there some way to know when the recovery process is complete (other than using the recovery API)?

File: docs/reference/ccr/remote-recovery.asciidoc [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+The following setting can be used to rate-limit the data transmitted during remote
```
Comment (lcawl) [inline docs/reference/ccr/remote-recovery.asciidoc]: We generally store these types of settings in the Elasticsearch Reference in pages like the ones linked here: https://www.elastic.co/guide/en/elasticsearch/reference/master/settings-xpack.html

I am happy to create that page if you agree.

Comment (tbrooks8) [reply docs/reference/ccr/remote-recovery.asciidoc]: Yeah that makes sense.

File: docs/reference/ccr/remote-recovery.asciidoc (hunk 9 above)
Comment (tbrooks8) [reply docs/reference/ccr/remote-recovery.asciidoc]: The recovery API is the primary. You could probably use the `cat indices` api to check if the index is green. You can also add a parameter to a follower request to wait until the process is completed. However, that is documented on the put follow request page.

**Comment by tbrooks8** [comment]
> Were you intending this new page to be added to the existing CCR documentation in the Stack Overview? 

Yes. Thanks.

File: docs/reference/ccr/remote-recovery.asciidoc (hunk 10 above)
Comment (lcawl) [reply docs/reference/ccr/remote-recovery.asciidoc]: I've added those changes.  You can preview the new page in the Elasticsearch Reference here:
https://doc-review-63bf6.firebaseapp.com/es2/ccr-settings.html 

I've also updated the Remove recovery page in the Stack Overview to link to that content, which you can preview here:
https://doc-review-63bf6.firebaseapp.com/x4/remote-recovery.html

**Review by lcawl — APPROVED** [review]
LGTM

**Comment by lcawl** [comment]
In case it helps, you can preview these new pages here: 
https://doc-review-63bf6.firebaseapp.com/es2/ccr-settings.html
https://doc-review-63bf6.firebaseapp.com/x4/remote-recovery.html

**Pull request merged**